code/modules/food_and_drinks/machinery/griddle.dm

```diff
@@ -79,6 +80,42 @@
 	else
 		return ..()
 
+/obj/machinery/griddle/item_interaction(mob/living/user, obj/item/item, list/modifiers, is_right_clicking)
+
+	if(item.atom_storage && !is_right_clicking)
+		var/obj/item/storage/tray = item
+		var/loaded = 0
+
+		if(!istype(item, /obj/item/storage/bag/tray))
+			// Non-tray dumping requires a do_after
+			to_chat(user, span_notice("You start dumping out the contents of [item] into [src]..."))
+			if(!do_after(user, 2 SECONDS, target = tray))
+				return
+
+		for(var/obj/tray_item in tray.contents)
+			if(!IS_EDIBLE(tray_item))
+				continue
+			if(contents.len >= max_items)
+				balloon_alert(user, "it's full!")
+				return TRUE
+			if(tray.atom_storage.attempt_remove(tray_item, src))
+				loaded++
+				AddToGrill(tray_item, user)
+		if(loaded)
+			to_chat(user, span_notice("You insert [loaded] items into \the [src]."))
+			update_appearance()
+		. = ..()
+		return
```

Comment:
```suggestion
			update_appearance()
		return
```
- Because we call parent up at the top, we don't need this parent call here anymore. 
- And because we did something successfully, `return ITEM_INTERACT_SUCCESS`.